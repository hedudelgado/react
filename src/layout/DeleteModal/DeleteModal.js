import React, { Fragment } from 'react';
import { string, func } from 'prop-types';
import Typography from '../Typography';
import Button from '../Button';

const DeleteModal = ({
  title,
  subtext,
  deleteFunc,
  buttonLabel,
  handleClose,
}) => (
  <Fragment>
    <Typography component="h5" mt="1">{title}</Typography>
    <Typography component="p" size="s">{subtext}</Typography>
    <Button data-test="delete-button" type="button" color="primary" className="mt2 mb2" onClick={deleteFunc} fullWidth>{buttonLabel}</Button>
    <Button data-test="cancel-button" type="button" color="default" variant="text" className="underline" onClick={handleClose}>Cancel changes</Button>
  </Fragment>
);

DeleteModal.propTypes = {
  title: string.isRequired,
  subtext: string.isRequired,
  deleteFunc: func.isRequired,
  handleClose: func.isRequired,
  buttonLabel: string.isRequired,
};

export default DeleteModal;
