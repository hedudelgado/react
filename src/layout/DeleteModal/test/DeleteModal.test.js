import React from 'react';
import { shallow } from 'enzyme';
import DeleteModal from '../DeleteModal';

describe('DeleteModal Component', () => {
  let wrapper;

  const props = {
    dictionary: {
      'guests.delete.title': 'Delete regular guest',
      'guests.delete.subtext': 'Are you sure you want to delete this guest?',
      'guests.delete.button.label': 'Delete guests',
      'guests.cancel.link': 'Cancel changes',
    },
    title: 'Delete guest',
    subtext: 'Are you sure you want to delete this guest?',
    deleteFunc: jest.fn(),
    buttonLabel: 'Delete guest',
    handleClose: jest.fn(),
    cancelLabel: 'Cancel changes',
  };

  beforeEach(() => {
    wrapper = shallow(<DeleteModal {...props} />);
  });

  describe('DeleteModal', () => {
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should execute the delete function when clicking the delete button', () => {
      wrapper.find('[data-test="delete-button"]').simulate('click');
      expect(props.deleteFunc).toHaveBeenCalled();
    });

    it('should execute the handleClose function when clicking cancel changes button', () => {
      wrapper.find('[data-test="cancel-button"]').simulate('click');
      expect(props.handleClose).toHaveBeenCalled();
    });
  });
});
