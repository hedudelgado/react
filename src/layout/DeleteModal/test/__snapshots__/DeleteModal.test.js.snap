// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal Component DeleteModal should match snapshot 1`] = `
<Fragment>
  <Typography
    component="h5"
    mt="1"
  >
    Delete guest
  </Typography>
  <Typography
    component="p"
    size="s"
  >
    Are you sure you want to delete this guest?
  </Typography>
  <Button
    className="mt2 mb2"
    color="primary"
    data-test="delete-button"
    fullWidth={true}
    href={null}
    onClick={[MockFunction]}
    submit={false}
    type="button"
    variant="button"
  >
    Delete guest
  </Button>
  <Button
    className="underline"
    color="default"
    data-test="cancel-button"
    fullWidth={false}
    href={null}
    onClick={[MockFunction]}
    submit={false}
    type="button"
    variant="text"
  >
    Cancel changes
  </Button>
</Fragment>
`;
